react-components/src/components/CadModelContainer/useApplyCadModelStyling.tsx

```diff
@@ -0,0 +1,125 @@
+import { CogniteCadModel, DefaultNodeAppearance, NodeAppearance, NodeCollection, NodeIdNodeCollection, TreeIndexNodeCollection, IndexSet } from "@cognite/reveal";
+import { useEffect } from "react";
+import { CadModelStyling, useReveal } from "../..";
+import { NodeStylingGroup, TreeIndexStylingGroup, modelExists } from "./CadModelContainer";
+import { useSDK } from "../RevealContainer/SDKProvider";
+import { CogniteClient } from "@cognite/sdk";
+import { isEqual } from "lodash";
+
+export const useApplyCadModelStyling = (model?: CogniteCadModel, modelStyling?: CadModelStyling) => {
+    const viewer = useReveal();
+    const sdk = useSDK();
+    
+    const defaultStyle = modelStyling?.defaultStyle ?? DefaultNodeAppearance.Default;
+    const styleGroups = modelStyling?.groups;
+
+    useEffect(() => {
+        if (!modelExists(model, viewer) || styleGroups === undefined) return;
+        
+        applyStyling(sdk, model, styleGroups);
+      }, [styleGroups, model]);
+    
+      useEffect(() => {
+        if (!modelExists(model, viewer)) return;
+          
+        model.setDefaultNodeAppearance(defaultStyle);
+        
+        return () => {
+          if (!modelExists(model, viewer)) {
+            return;
+          }
+            
+          model.setDefaultNodeAppearance(DefaultNodeAppearance.Default);
```

Comment:
I am not 100% convinced that we actually need to do this. What does not doing this cause an issue for?

Reply:
With having `defaulStyle` defined as in the component, i.e. defaulting to actual  `DefaultNodeAppearance.Default` I think we don't need.

Reply:
It's just if we want it to look similar to how `styleGroups` are defined, i.e. without defaulting to any value, we can have it as is. Whatever is fine by me, removed it for now.